Newsletters: add 325 (2024-10-18)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Client/services @bitschmidty 
- [x] Notable merges @Gustavojfe

bitschmidty commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,64 @@
+
+## News
+
+_No significant news this week was found in any of our [sources][optech sources]._
```
https://delvingbitcoin.org/t/ln-summit-2024-notes-summary-commentary/1198 just dropped, FYI, if there is time

harding replied:
Added!

vostrnad commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,117 @@
+  [bech32m][topic bech32] addresses.
+
+- **Dana wallet released:**
+  [Dana wallet][dana wallet github] is a [silent payment][topic silent payments] wallet focused the
```
```suggestion
  [Dana wallet][dana wallet github] is a [silent payment][topic silent payments] wallet focused on the
```

vostrnad commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,117 @@
+  donation use case. The developers recommend using [signet][topic signet] and
+  also run a signet [faucet][dana wallet faucet].
+
+- **Bitcoin script editor and visualizer:**
+  The Saving Satoshi [script editor][saving satoshi editor] allows for visually
+  building and executing Bitcoin scripts.
```
I am not sure we should include this just yet, I've played with the editor for a few minutes and already found serious bugs in the script interpreter. For example, `OP_0 OP_SHA256` does not hash the empty byte array but the string "0". Additionally, only around half the existing opcodes are implemented, for example most arithmetic and stack opcodes are missing. And there are other issues, for example cleanstack is enforced without explaining why the script "failed".

bitschmidty replied:
This is good feedback. Will remove for this month.

vostrnad commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,117 @@
+  payment batching], [RBF][topic rbf] support, and [fee estimation][topic fee estimation].
+
+- **DATUM protocol announced:**
+  The [DATUM mining protocol][datum docs] the allows miners to build candidate blocks as part
```
```suggestion
  The [DATUM mining protocol][datum docs] allows miners to build candidate blocks as part
```

vostrnad commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,117 @@
+
+- [BDK 1.0.0-beta.5][] is a release candidate (RC) of this library for
+  building wallets and other Bitcoin-enabled applications.  This latest
+  RC "enables RBF by default, updates the bdk_esplora client to retry
```
```suggestion
  RC "enables RBF by default, it also updates the bdk_esplora client to retry
```
Or perhaps:
```suggestion
  RC "enables RBF by default [and] updates the bdk_esplora client to retry
```

shesek commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,198 @@
+    [#261][news261 simcom]).  Allowing either side to propose updates can
+    result in both side proposing updates simultaneously, which is
+    difficult to reason about and can lead to accidental channel force
+    closures when.  The alternative is for only one party to be in
```
"when" appears out of place

LarryRuane replied:
I think it should just be removed.

Gustavojfe commented:
just added merge summaries.

jirijakes commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,231 @@
+
+  - **State update protocol:** a proposal was discussed to convert LN's
+    current state update protocol from allowing either side to propose an
+    update at any time to only allowing a one party at a time to propose
```
```suggestion
    update at any time to only allowing one party at a time to propose
```

bitschmidty commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,235 @@
+[zmnscpxj superscalar]: https://delvingbitcoin.org/t/superscalar-laddered-timeout-tree-structured-decker-wattenhofer-factories/1143
+[news309 feasible]: /en/newsletters/2024/06/28/#estimating-the-likelihood-that-an-ln-payment-is-feasible
+[news310 mining]:/en/newsletters/2024/07/05/#bitcoin-core-30200
+[news323 fees]: /en/newsletters/2024/10/04/#eclair-2861
```
these two 323 references are reversed

LarryRuane commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,231 @@
+    update at any time to only allowing a one party at a time to propose
+    updates (see Newsletters [#120][news120 simcom] and
+    [#261][news261 simcom]).  Allowing either side to propose updates can
+    result in both side proposing updates simultaneously, which is
```
```suggestion
    result in both sides proposing updates simultaneously, which is
```

LarryRuane commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,231 @@
+
+- [LDK #3235][] adds a `last_local_balance_msats` field to the
+  `ChannelForceClosed` event, which gives the local balance of a node in
+  millisatoshis (msats) just before the channel was force closed, allowing users
```
```suggestion
  millisatoshis (msats) just before the channel was force-closed, allowing users
```

LarryRuane commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,231 @@
+
+- [LND #8183][] adds the optional `CloseTxInputs` field to the
+  `chanbackup.Single` structure in the [static channel backup][topic static
+  channel backups] (SCB) file, to store the inputs required to generate force
+  close transactions. This allows users to manually retrieve funds when a peer
```
```suggestion
  channel backups] (SCB) file, to store the inputs required to generate force-close
  transactions. This allows users to manually retrieve funds when a peer
```

LarryRuane commented on _posts/en/newsletters/2024-10-18-newsletter.md:
```diff
@@ -0,0 +1,231 @@
+  `chanbackup.Single` structure in the [static channel backup][topic static
+  channel backups] (SCB) file, to store the inputs required to generate force
+  close transactions. This allows users to manually retrieve funds when a peer
+  is offline using the `chantools scbforceclose` command as a last resort
```
```suggestion
  is offline using the `chantools scbforceclose` command as a last-resort
```

harding commented:
Made edits for all feedback, thanks @bitschmidty @vostrnad @shesek @jirijakes @LarryRuane !  Added lede, checked for updated releases/RCs (none found), added topic entries, and reviewed excellent sections from @bitschmidty and @Gustavojfe.  Thanks everyone!

bitschmidty reviewed (commented):
ACK lede, topics. Squashed. Thank you all! 🚀
